File: iosApp/iosApp/data/realm/RealmDb.swift
//
//  RealmDb.swift
//  CallMed_ios
//
//  Created by Mihail on 23.06.2023.
//

import Foundation
//import RealmSwift

//class RealmDb {
//    static let shared = RealmDb()
//    private init() { }
//    
//    
//   // let config = RealmConfiguration.create(schema = setOf(MessageRoomItem::class))
//    //let realm = try! Realm()
//    
//    var latchWrite = false
//    
//    func deleteMessage(_ item: MessageRoomItem){
//        let realm = try! Realm()
//        
//        try! realm.write {
//            let tmpItem: MessageRoomItem? = realm.objects(MessageRoomItem.self).where{$0.id_message == item.id_message}.first
//            if let tmpItmem = tmpItem {
//                realm.delete(tmpItmem)
//            }
//        }
//        
//        
////        try! realm.write { query<MessageRoomItem>("idMessage == $0", item.idMessage).first().find()
////            ?.also { delete(it) }
////        }
//        
//    }
//    
//    func isPossibleDeleteCheckMsgUserAfterSelect(_ item: MessageRoomItem) -> Bool {
//        let realm = try! Realm()
//        
//        let listItems: Results<MessageRoomItem> = realm.objects(MessageRoomItem.self).where{$0.id_room == item.id_room!}
//        if(listItems.count == 0){
//            return true
//        }
//
//        let listItems2: Results<MessageRoomItem> =  listItems.where{$0.id_message > item.id_message}
//        
//        if(listItems2.count == 0){
//            return true
//        }
//
//        for i in listItems2 {
//            if(i.otpravitel == "kl"){
//                return false
//            }
//        }
//
//        return true
//    }
//    
//    func addListMessages(_ list: [MessageRoomItem]) -> [MessageRoomItem]{
//        let realm = try! Realm()
//        
//        latchWrite = true
//        
//        var listNewM: [MessageRoomItem] = []
//        
//        for i in list {
//            let item: MessageRoomItem? = realm.objects(MessageRoomItem.self).where{$0.id_message == i.id_message!}.first
//            if(item == nil){
//                listNewM.append(i)
//            }
//        }
//        
//        try! realm.write {
//            for i in listNewM {
//                //let tmp = i.idAsString()
//                realm.add(i)
//            }
//            
//            latchWrite = false
//        }
//        
//        return listNewM
//    }
//    
//    func getAllMessageByIdRoom(_ idRoom: String) -> [MessageRoomItem]{
//        let realm = try! Realm()
//        
//        // RealmResults<MessageRoomItem>
//        let incompleteItems: Results<MessageRoomItem> = realm.objects(MessageRoomItem.self).where{$0.id_room == idRoom}
//
//        if(incompleteItems.count==0){
//            return []
//        }else{
//            var newArr:[MessageRoomItem] = []
//            //let oldArray = incompleteItems.toArray()
//            
//            for i in incompleteItems{
//                let tmp = RealmHelper.DetachedCopy(itemRealm: i)
//                
//                if let tmp = tmp {
//                    newArr.append(tmp)
//                }
//                
//            }
//            return newArr
//        }
//    }
//    
//    func getMaxIdMessageByIdRoom(_ idRoom: String) -> Int {
//        let realm = try! Realm()
//        
//        let maxIdMessage: Int? = realm.objects(MessageRoomItem.self).where{$0.id_room == idRoom}.max(ofProperty: "id_message") as Int?
//        //let maxIdMessage: Int? = realm.query<MessageRoomItem>("idRoom == $0", idRoom).max("idMessage", Int::class).find()
//        return maxIdMessage ?? 0
//    }
//
//}
//
//
//
//extension Results {
//    func toArray() -> [Element] {
//      return compactMap {
//        $0
//      }
//    }
// }

File: androidApp/src/main/java/com/medhelp/callmed2/data/db/RealmDb.kt
//package com.medhelp.callmed2.data.db
//
//import com.medhelp.callmedcmm2.model.chat.MessageRoomItem
//import io.realm.kotlin.Realm
//import io.realm.kotlin.RealmConfiguration
//import io.realm.kotlin.query.RealmResults
//import io.realm.kotlin.ext.query
//import kotlinx.coroutines.GlobalScope
//import kotlinx.coroutines.launch
//
//object RealmDb {
//    val config = RealmConfiguration.create(schema = setOf(MessageRoomItem::class))
//    val realm: Realm = Realm.open(config)
//
//    fun deleteMessage(item: MessageRoomItem){
//        GlobalScope.launch {
//            realm.write {
//                query<MessageRoomItem>("idMessage == $0", item.idMessage)
//                    .first()
//                    .find()
//                    ?.also { delete(it) }
//            }
//        }
//    }
//
//    fun getAllMessageByIdRoom(idRoom: String): List<MessageRoomItem>{
//        val incompleteItems: RealmResults<MessageRoomItem> = realm.query<MessageRoomItem>("idRoom == $0", idRoom).find()
//
//        if(incompleteItems.size==0)
//            return listOf()
//        else
//            return realm.copyFromRealm(incompleteItems)
//    }
//
//    fun getMaxIdMessageByIdRoom(idRoom: String) : Int {
//        //val minIdMessage: Int? = realm.query<MessageRoomItem>("idRoom == $0", idRoom).min("idMessage", Int::class).find()
//        val maxIdMessage: Int? = realm.query<MessageRoomItem>("idRoom == $0", idRoom).max("idMessage", Int::class).find()
//        return maxIdMessage ?: 0
//    }
//
//    var latchWrite = false
//    fun addListMessages(list: List<MessageRoomItem>): List<MessageRoomItem>{
//        latchWrite = true
//
//        var listNewM: MutableList<MessageRoomItem> = mutableListOf()
//
//        for(i in list){
//            val item: MessageRoomItem? = realm.query<MessageRoomItem>("idMessage == $0", i.idMessage!!).first().find()
//            if(item == null){
//                listNewM.add(i)
//            }
//        }
//
//        GlobalScope.launch {
//            realm.write {
//                for (i in listNewM) {
//                    copyToRealm(i)
//                }
//
//                latchWrite = false
//            }
//        }
//
//
//        return listNewM
//    }
//
//    fun isPossibleDeleteCheckMsgUserAfterSelect(item: MessageRoomItem) : Boolean {
//        val listItems: RealmResults<MessageRoomItem> = realm.query<MessageRoomItem>("idRoom == $0", item.idRoom!!).find()
//        if(listItems.size==0)
//            return true
//
//        val listItems2: RealmResults<MessageRoomItem> =  listItems.query("idMessage > $0", item.idMessage).find()
//        if(listItems2.size==0)
//            return true
//
//        for(i in listItems2){
//            if(i.otpravitel == "kl")
//                return false
//        }
//
//        return true
//    }
//}